### PR TITLE
Create OpenAPIV3 schema for vsphere-cpi package and generate package with the schema, but for version 1.23.0-alpha.1

### PR DESCRIPTION
**Pull request by HanFa**
## What this PR does / why we need it

This is a follow-up PR of https://github.com/vmware-tanzu/community-edition/pull/2805. 

It adds the schema.yaml to the latest version vsphere-cpi and updates the package manifest.

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Create OpenAPIV3 schema for vsphere-cpi package and generate package with the schema, but for version 1.23.0-alpha.1
```

## Which issue(s) this PR fixes

Fixes: #

## Describe testing done for PR

vsphere-cpi unit test.

### CODE REVIEW

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    #@schema/desc "The namespace of secret that stores CPI configuration"
+    secretNamespace: "kube-system"
+  #@schema/desc "HTTP proxy setting"
+  http_proxy: ""
+  #@schema/desc "HTTPS proxy setting"
+  https_proxy: ""
+  #@schema/desc "No-proxy setting"
+  no_proxy: ""
+  #@schema/desc "Used in vsphereParavirtual mode, defines the Cluster API versions. Default: run.tanzu.vmware.com/v1alpha2."
+  clusterAPIVersion: "run.tanzu.vmware.com/v1alpha2"
```
Comment (blc1996) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml:87]: Seems we have `run.tanzu.vmware.com/v1alpha3` available in [t-f](https://github.com/vmware-tanzu/tanzu-framework/tree/main/apis/run/v1alpha3), please confirm the version used here.

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  #@schema/desc "Used in vsphereParavirtual mode, defines the Cluster kind. Default: Cluster."
+  clusterKind: "Cluster"
+  #@schema/desc "Used in vsphereParavirtual mode, defines the Cluster name. Default: test-cluster."
+  clusterName: "test-cluster"
```
Comment (blc1996) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml:91]: The default clusterName could be empty.

Comment (blc1996) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml:91]: Also, we can add some more assertions if param like `clusterName` is mandatory for paravirtualized mode.

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml (hunk 1 above)
Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml:87]: good catch, let's use latest the `v1alpha3`

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml (hunk 2 above)
Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml]: @blc1996 I've changed the clusterName and clusterUID default to be empty. Besides, some assertions are added as well for them to be non-empty in the paravirtual mode.

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml (hunk 1 above)
Comment (lubronzhan) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml]: The version in `t-f` is for TKR, while the clusterAPIVersion is the version for Cluster API CRD. 
For uTKG we should use whatever Cluster CRD version that the current Cluster has
https://github.com/kubernetes-sigs/cluster-api/tree/main/api/v1beta1

Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/schema.yaml]: changed to `v1beta1`

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/README.md [hunk 3]
```diff
@@ -42,7 +44,7 @@ None
 |`vsphereCPI.nsxt.rootCAData` | Optional | The certificate authority for the server certificate for locally signed certificates. Default: `""`. |
 | `vsphereCPI.nsxt.secretName` | Optional | The name of secret that stores CPI configuration. Default: `cloud-provider-vsphere-nsxt-credentials`. |
 | `vsphereCPI.nsxt.secretNamespace`| Optional | The namespace of secret that stores CPI configuration. Default: `True`. |
-| `vsphereCPI.clusterAPIVersion`| Optional for `vsphereParavirtual` | Used in `vsphereParavirtual` mode, defines the Cluster API versions. Default: `run.tanzu.vmware.com/v1alpha2` |
+| `vsphereCPI.clusterAPIVersion`| Optional for `vsphereParavirtual` | Used in `vsphereParavirtual` mode, defines the Cluster API versions. Default: `run.tanzu.vmware.com/v1beta1` |
```
Comment (lubronzhan) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/README.md:47]: Could you also update the group version to align with https://github.com/kubernetes-sigs/cluster-api/blob/main/api/v1beta1/groupversion_info.go#L29

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/test/vsphere_cpi_test.go [hunk 4]
```diff
@@ -165,7 +165,7 @@ vsphereCPI:
 				ownerRefConfigMap := findConfigMapByName(configMaps, "ccm-owner-reference")
 				ownerRef, exists := ownerRefConfigMap.Data["owner-reference"]
 				Expect(exists).To(BeTrue())
-				Expect(ownerRef).To(Equal("{\"apiVersion\": \"run.tanzu.vmware.com/v1alpha2\",\n\"kind\": \"Cluster\",\n\"name\": \"tkg-cluster\",\n\"uid\": \"57341fa8-0983-472f-b744-00cf724dd307\"}\n"))
+				Expect(ownerRef).To(Equal("{\"apiVersion\": \"run.tanzu.vmware.com/v1beta1\",\n\"kind\": \"Cluster\",\n\"name\": \"tkg-cluster\",\n\"uid\": \"57341fa8-0983-472f-b744-00cf724dd307\"}\n"))
```
Comment (lubronzhan) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/test/vsphere_cpi_test.go:168]: Same here

**Review by blc1996 — APPROVED** [review]
LGTM

**Review by lubronzhan — APPROVED** [review]
/lgtm

**Pull request merged**